video/decode/vd_lavc.c

```diff
@@ -952,6 +958,40 @@ static enum AVPixelFormat get_format_hwdec(struct AVCodecContext *avctx,
     return select;
 }
 
+static int get_buffer2_direct(AVCodecContext *avctx, AVFrame *pic, int flags)
+{
+    struct dec_video *vd = avctx->opaque;
+    struct vo_buffer_data vbufd = { 0 };
+
+    struct mp_image img;
+    mp_image_copy_fields_from_av_frame(&img, pic);
+    vbufd.par            = &img.params;
+    vbufd.aligned_width  = img.params.w;
+    vbufd.aligned_height = img.params.h;
+    avcodec_align_dimensions2(avctx, &vbufd.aligned_width, &vbufd.aligned_height,
+                              vbufd.stride_alignment);
+
+    int ret = vd->vo->driver->get_buffer(vd->vo, &vbufd);
+    if (ret < 0)
+        return AVERROR(ENOMEM);
+    else if (ret > 0)
+        return avcodec_default_get_buffer2(avctx, pic, flags);
```

Comment:
Its needed to be able to turn off direct rendering from inside the VO using an option. Though I think it might actually be better to have a global option for all VOs.